lib/auth/trustedcluster.go

```diff
@@ -67,8 +68,15 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, trustedCluster types.
 	}
 
 	// change state
+
+	if exists && !cmp.Equal(existingCluster.GetRoleMap(), trustedCluster.GetRoleMap()) {
+		if err := a.checkLocalRoles(ctx, trustedCluster.GetRoleMap()); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		a.updateUserCARoleMap(ctx, existingCluster, trustedCluster)
+	}
 	switch {
-	case exists == true && enable == true:
+	case exists == true && enable == true && existingCluster.GetEnabled() != trustedCluster.GetEnabled():
```

Comment:
Why did this condition change? Wouldn't this prevent user from upserting the trusted cluster if it's already enabled?

Same for the "exists == true && enable == false" case below.